pkg_resources/__init__.py

```diff
@@ -2825,7 +2825,7 @@ def activate(self, path=None, replace=False):
         if path is None:
             path = sys.path
         self.insert_on(path, replace=replace)
-        if path is sys.path:
+        if path is sys.path and self.location:
```

Comment:
I wonder if it is possible that `self.location` is an empty string...
The empty string is a valid entry for `sys.path` (and holds special meaning).

Reply:
Just in case:
```suggestion
        if path is sys.path and self.location is not None:
```